expression/builtin_string.go

```diff
@@ -277,89 +276,80 @@ type leftFunctionClass struct {
 }
 
 func (c *leftFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinLeftSig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt)
+	bf.tp.Flen = args[0].GetType().Flen
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
`err` checking should be just after the function `newBaseBuiltinFuncWithTp` called